xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/v3/BinarySerializationTests.java

```diff
@@ -384,6 +387,184 @@ void serializeTrustLine() throws JsonProcessingException {
     assertSerializesAndDeserializes(trustSet, expectedBinary);
   }
 
+  @Test
+  void serializeTrustSetWithMaxValue() throws JsonProcessingException {
+    TrustSet trustSet = TrustSet.builder()
+      .account(Address.of("rBZwgHHifSeFmf2btuubr6YUhZ3FfW47XE"))
+      .fee(XrpCurrencyAmount.ofDrops(10))
+      .sequence(UnsignedInteger.valueOf(5))
+      .limitAmount(IssuedCurrencyAmount.builder()
+        .currency("7872706C346A436F696E00000000000000000000")
+        .issuer(Address.of("rwk7evAoUG1YoH1k3KVps3Zm1MtHuwHadj"))
+        .value(IssuedCurrencyAmount.MAX_VALUE)
+        .build())
+      .signingPublicKey("EDD299D60BCE7980F6082945B5597FFFD35223F1950673BFA4D4AED6FDE5097156")
+      .build();
+
+    String expectedBinary = "1200142280000000240000000563EC6386F26FC0FFFF7872706C346A436F696E00000000000000000000" +
+      "6AF21A4FCA7AF4F62AA88E72CA923CDAE72FBC1668400000000000000A7321EDD299D60BCE7980F6082945B5597FFFD35223F19506" +
+      "73BFA4D4AED6FDE5097156811473C6A5937EB779F5CF33600DB0ADB8556A497318";
+    String transactionJson = objectMapper.writeValueAsString(trustSet);
+    String transactionBinary = binaryCodec.encode(transactionJson);
+    assertThat(transactionBinary).isEqualTo(expectedBinary);
+
+    String decodedBinary = binaryCodec.decode(transactionBinary);
+    TrustSet deserialized = objectMapper.readValue(
+      decodedBinary,
+      objectMapper.getTypeFactory().constructType(TrustSet.class)
+    );
+
+    assertThat(deserialized).usingRecursiveComparison().ignoringFields("limitAmount").isEqualTo(trustSet);
+    BigDecimal deserializedValue = new BigDecimal(deserialized.limitAmount().value());
+    BigDecimal originalValue = new BigDecimal(trustSet.limitAmount().value());
+    assertThat(deserializedValue.toPlainString()).isEqualTo(originalValue.toPlainString());
+  }
+
+  @Test
+  void serializeTrustSetWithMaxValueMinusOnePowerOfTen() throws JsonProcessingException {
+    TrustSet trustSet = TrustSet.builder()
+      .account(Address.of("r4V5fhT5KPCdhUanXCMpPUHiM4s6bXDVDH"))
+      .fee(XrpCurrencyAmount.ofDrops(10))
+      .sequence(UnsignedInteger.valueOf(5))
+      .limitAmount(IssuedCurrencyAmount.builder()
+        .currency("7872706C346A436F696E00000000000000000000")
+        .issuer(Address.of("rUcccYgErfgMifUHYDHed4ibTArjqfkU5R"))
+        .value(new BigDecimal(IssuedCurrencyAmount.MAX_VALUE).scaleByPowerOfTen(-1).toString())
```

Comment:
These exponent +-1 tests are useful (so we should keep them) but I was originally thinking just MAX-1 and MAX+1 to detect an "off-by-one" error and detect what happens on overflow. Could we add those too?

Reply:
We could add those, but XRP Ledger only understands a precision of 16. Our binary codec throws a different kind of error of the precision is off. Since I was trying to test the change in exponent bounds, I figured this was more relevant.

Reply:
Yeah, agree - I think they're both important.

https://twitter.com/codinghorror/status/506010907021828096